### ISSUE TITLE
Terraform template: unknown variable accessed

### ISSUE THREAD

**Issue by JeanMertz** [hashicorp/terraform#2236]
It seems that if I add new template variables (and remove others) in between runs, Terraform is confused and can't run a `plan` anymore:

```
Error refreshing state: 1 error(s) occurred:

* 1 error(s) occurred:

* failed to render user-data.yml: 132:65: unknown variable accessed: api_subdomain
```

**Comment by josephholsten** [comment]
dup #2013

**Comment by josephholsten** [comment]
@phinze should be closed by #2386

**Comment by phinze** [comment]
Yep! Fix landed in master and will come out with 0.6. :+1:
